src/app/components/plugin-uiframe/plugin-uiframe.component.ts

```diff
@@ -291,6 +291,9 @@ export class PluginUiframeComponent implements OnChanges, OnDestroy {
     }
 
     private loadImplementations(): void {
+        const allowedContentTypes: Set<string> = new Set(["application/qasm", "application/qiskit"]);
+        const contentTypeToType = (contentType: string) => contentType.split("/")[1];
```

Comment:
This is not a Map (java equivalent is a HashMap) but a mapping function...
The strategy of this function should only be used as a fallback if no explicit mapping is configured in a Map.